query/src/main/java/io/keen/client/java/MultiAnalysis.java

```diff
@@ -0,0 +1,172 @@
+
+
+    protected MultiAnalysis(Builder builder) {
+        super(builder);
+
+        this.analyses = builder.subAnalyses;
+    }
+
+    @Override
+    URL getRequestURL(RequestUrlBuilder urlBuilder, String projectId)
+            throws KeenQueryClientException {
+        return urlBuilder.getAnalysisUrl(projectId, KeenQueryConstants.MULTI_ANALYSIS);
+    }
+
+    @Override
+    Map<String, Object> constructRequestArgs() {
+        // First get the more general top-level args for this analysis.
+        Map<String, Object> multiAnalysisArgs = super.constructRequestArgs();
+
+        // validateParams() makes sure this.analyses is properly configured.
+        Map<String, Map<String, Object>> analysesArgs = new HashMap<String, Map<String, Object>>();
+
+        // Add each of the sub-analyses as "label": { ...sub-analysis args... }
+        for (SubAnalysis subAnalysis : this.analyses) {
+            Map<String, Object> subAnalysisArgs = subAnalysis.constructParameterRequestArgs();
+
+            analysesArgs.put(subAnalysis.getLabel(), subAnalysisArgs);
+        }
+
+        // Add the top-level "analyses" key
+        multiAnalysisArgs.put(KeenQueryConstants.ANALYSES, analysesArgs);
+
+        return multiAnalysisArgs;
+    }
+
+    @Override
+    protected void validateParams() {
+        super.validateParams();
+
+        if (null == this.analyses || this.analyses.isEmpty()) {
+            throw new IllegalArgumentException("MultiAnalysis requires that 'analyses' be set.");
+        }
+    }
+
+    /**
+     * This builder class helps configure the required and optional parameters that are appropriate
+     * for a MultiAnalysis.
+     */
+    public static class Builder extends CollectionAnalysis.Builder<Builder> {
+        // required
+        private Collection<SubAnalysis> subAnalyses;
+
+
+        /**
+         * Get the collection of sub-analyses for this multi-analysis.
+         *
+         * @return The sub-analyses.
+         */
+        public Collection<SubAnalysis> getSubAnalyses() { return this.subAnalyses; }
+
+
+        // Configurable properties added by this builder.
+
+        /**
+         * Set the collection of sub-analyses. Existing sub-analyses will be discarded.
+         *
+         * @param subAnalyses The new analysis arguments to replace the discarded ones, if any.
+         */
+        public void setSubAnalyses(Collection<? extends SubAnalysis> subAnalyses) {
+            this.subAnalyses = null;
+
+            // Client code may just be clearing all the analyses.
+            if (null != subAnalyses) {
+                withSubAnalyses(subAnalyses);
+            }
+        }
+
+        /**
+         * Adds a collection of sub-analyses.
+         *
+         * @param subAnalyses The new analysis arguments to add to the existing analyses.
+         * @return This instance (for method chaining).
+         */
+        public Builder withSubAnalyses(Collection<? extends SubAnalysis> subAnalyses) {
```

Comment:
Suggestion: Similar issue here as with the `setSteps`/`withSteps` in the funnel case. I think it generally improves API predictability if `withXxx(...)` can be assumed to always be implemented as `setXxx(...); return this;` (or `addXxx(...); return this;`, if that makes more sense semantically - like in the singular `withSubAnalysis` case below).